tests/test_api/test_validators/test_generate_teams_validator/test_generate_teams_validator.py

```diff
@@ -217,3 +267,45 @@ def test_errors__team_generation_options(self):
             )
             validator.validate()
         self.assertEqual(e.exception.code, "Team ids must be unique.")
+
+    def test_errors__algorithm_options__random_algorithm(self):
+        invalid_field_random_algorithm_data = copy.deepcopy(
+            self.valid_random_algorithm_data
+        )
+        invalid_field_random_algorithm_data["algorithm_options"][
+            "max_project_preferences"
+        ] = 3
+        with self.assertRaises(SchemaError) as e:
+            validator = GenerateTeamsValidator(invalid_field_random_algorithm_data)
+            validator.validate()
+        self.assertIn("Wrong key 'max_project_preferences' in ", e.exception.code)
+
+    def test_errors__algorithm_options__weight_algorithm(self):
```

Comment:
idk if I like this naming scheme, I think the "test_{name of function}__{explain test briefly}" is a better idea here still

Reply:
kind of a nitpick but also I think we should have test names just give context on what the problem is without having to read the whole test